unity/Group55TestProjekt/Assets/Scripts/model/IntraCellular/ForwardInternals.cs

```diff
@@ -100,9 +107,14 @@ public State GetInternalState()
 
     public IPointAdapter GetNextLocation()
     {
-        currentIteration++;
-        if (currentIteration > iterations + 1)
-            currentIteration = iterations + 1;
-        return positions[currentIteration];
+        IPointAdapter point = positions[currentIteration];
+        currentIteration = currentIteration+1 > iterations ? iterations : currentIteration+1;
+       
+        return point;
+    }
+
+    public float GetAngle()
+    {
+        return initalAngel;
```

Comment:
Angle

Reply:
nah it is an angel we need the power of God for the code to run

Reply:
*resolved*